api/v1alpha1/authorization_types.go

```diff
@@ -47,17 +50,55 @@ type Principal struct {
 	// ClientCIDRs are the IP CIDR ranges of the client.
 	// Valid examples are "192.168.1.0/24" or "2001:db8::/64"
 	//
+	// If multiple CIDR ranges are specified, one of the CIDR ranges must match
+	// the client IP for the rule to match.
+	//
 	// The client IP is inferred from the X-Forwarded-For header, a custom header,
 	// or the proxy protocol.
 	// You can use the `ClientIPDetection` or the `EnableProxyProtocol` field in
 	// the `ClientTrafficPolicy` to configure how the client IP is detected.
 	// +kubebuilder:validation:MinItems=1
 	ClientCIDRs []CIDR `json:"clientCIDRs"`
 
-	// TODO: Zhaohuabing the MinItems=1 validation can be relaxed to allow empty list
-	// after other principal types are supported. However, at least one principal is required
+	// JWTClaims are the claims in a JWT token.
+	//
+	// If multiple claims are specified, all claims must match for the rule to match.
+	// For example, if there are two claims: one for the audience and one for the issuer,
+	// the rule will match only if both the audience and the issuer match.
+	// Note: in order to use JWT claims for authorization, you must configure the
+	// JWT authentication in the same `SecurityPolicy`.
+	// +optional
+	// +notImplementedHide
+	JWTClaims []JWTClaim `json:"jwtClaims"`
 }
 
+// JWTClaim specifies a claim in a JWT token.
+type JWTClaim struct {
+	// Type is the type of the claim.
+	// Valid values are "String" and "StringArray".
+	// For example, sub is a string claim, and groups is a string array claim.
+	// +kubebuilder:validation:Enum=String;StringArray
+	// +kubebuilder:default=String
+	// +unionDiscriminator
+	Type JWTClaimType `json:"type"`
+
+	// Name is the name of the claim.
+	Name string `json:"name"`
+
+	// Values are the values that the claim must match.
+	// If the claim is a string type, the specified value must match exactly.
+	// If the claim is a string array type, the specified value must match one of the values in the array.
+	// If multiple values are specified, one of the values must match for the rule to match.
+	Values []string `json:"values"`
```

Comment:
how would this work if values are boolean or integer ?

Reply:
All the well-known claims that can be used for authorization, such as iss, sub, aud, email, name, roles, scope, and groups,  are either string or string array. Are there any other claims that could be used as identities but are of a different type?

We could consider making this API more flexible to also accommodate boolean and integer types, but it might introduce unnecessary complexity, and it's unclear if such an extension would be essential.  

@envoyproxy/gateway-maintainers @envoyproxy/gateway-reviewers please weigh in as you may have use cases for the jwt claims as authorization identities.

Reply:
`email_verified` is boolean claim. That might be the only one that people could use for authorization

Reply:
can we represent claim values as string, and when matching convert a bool or int to string and compare ?

Reply:
> can we represent claim values as string, and when matching convert a bool or int to string and compare ?

Agree. I'll update the API comments to reflect this.